cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTable.java

```diff
@@ -65,19 +64,18 @@
 
   public static final String DELTA_WRITE = "d";
   private final HTable hTable;
-  private final String hTableName;
+  private final String tableName;
 
   private Transaction tx;
 
   private final TransactionCodec txCodec;
 
-  public HBaseTable(String name, ConflictDetection level, Configuration hConf, boolean enableReadlessIncrements)
-    throws IOException {
-    super(name, level, enableReadlessIncrements);
-    hTableName = HBaseTableUtil.getHBaseTableName(name);
-    HBaseTableUtil tableUtil = new HBaseTableUtilFactory().get();
-    TableId tableId = TableId.from(name);
-    HTable hTable = tableUtil.getHTable(hConf, tableId);
+  public HBaseTable(String tableName, ConflictDetection level, Configuration hConf,
+                    HBaseTableUtil tableUtil, boolean enableReadlessIncrements) throws IOException {
+    super(tableName, level, enableReadlessIncrements);
+    this.tableName = tableName;
+    TableId tableId = TableId.from(tableName);
```

Comment:
Yes. Bhooshan has that in his PR.
https://github.com/caskdata/cdap/pull/1495/files#diff-3940118a7982e1cceeaa023c99b1b35f